infrastructure/src/main/java/org/corfudb/infrastructure/log/CompactionMetadata.java

```diff
@@ -0,0 +1,157 @@
+            }
+
+            if (logData.getType() != DataType.DATA) {
+                log.debug("updateTotalPayloadSize: LogData[{}] is not DATA or COMPACTED " +
+                        "type, skip recording stats. Type: {}",
+                        logData.getGlobalAddress(), logData.getType());
+                continue;
+            }
+
+            // If this LogData is sent from client, it should have data field.
+            if (logData.getData() != null) {
+                totalPayloadSize += logData.getData().length;
+                continue;
+            }
+
+            if (logData.getLastKnownSize() != LogData.NOT_KNOWN) {
+                totalPayloadSize += logData.getLastKnownSize();
+                continue;
+            }
+
+            try (ILogData.SerializationHandle sh = logData.getSerializedForm()) {
+                totalPayloadSize += logData.getSizeEstimate();
+            }
+        }
+    }
+
+    /**
+     * Update garbage size for this segment.
+     *
+     * @param garbageEntries garbage entries appended to log,
+     *                       which contains garbage size information
+     */
+    void updateGarbageSize(Collection<SMRGarbageEntry> garbageEntries) {
+        for (SMRGarbageEntry garbageEntry : garbageEntries) {
+            updateBoundedGarbageSize(garbageEntry);
+            updateTotalGarbageSize(garbageEntry);
+        }
+    }
+
+    private void updateTotalGarbageSize(SMRGarbageEntry garbageEntry) {
+        totalGarbageSize += garbageEntry.getGarbageSize();
+    }
+
+    private void updateBoundedGarbageSize(SMRGarbageEntry garbageEntry) {
+        if (compactionUpperBound != nextCompactionUpperBound) {
+            compactionUpperBound = nextCompactionUpperBound;
+            boundedGarbageSize = totalGarbageSize;
+        }
+        boundedGarbageSize += garbageEntry.getGarbageSizeUpTo(compactionUpperBound);
+    }
+
+    /**
+     * Get the total size of the garbage in this segment in MB.
+     *
+     * @return total size of the garbage payloads in MB
+     */
+    double getTotalGarbageSizeMB() {
+        return (double) totalGarbageSize / 1024 / 1024;
+    }
+
+    /**
+     * Get the size of the bounded garbage in this segment in MB.
+     *
+     * @return total size of the garbage payloads in MB
+     */
+    double getBoundedGarbageSizeMB() {
+        return (double) boundedGarbageSize / 1024 / 1024;
+    }
+
+    /**
+     * Get the garbage ratio of this segment, which is defined by
+     * total size of garbage payload  / total payload size.
+     *
+     * @return garbage ratio of this segment
+     */
+    double getGarbageRatio() {
+        return (double) totalGarbageSize / totalPayloadSize;
+    }
+
+    /**
+     * Get the bounded garbage ratio of this segment, which is defined
+     * by size of bounded garbage payload  / total payload size.
+     *
+     * @return bounded garbage ratio of this segment
+     */
+    double getBoundedGarbageRatio() {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use explicit scoping instead of the default package private level](https://app.codacy.com/app/annymartinez.m/CorfuDB/pullRequest?prid=4651450)